warcprox/mitmproxy.py

```diff
@@ -163,13 +164,17 @@ def __init__(
                 self.fp, proxy_client, digest_algorithm, url=url)
         self.fp = self.recorder
 
-    def begin(self):
+    def begin(self, extra_response_headers={}):
         http_client.HTTPResponse.begin(self)  # reads status line, headers
 
         status_and_headers = 'HTTP/1.1 {} {}\r\n'.format(
                 self.status, self.reason)
         self.msg['Via'] = via_header_value(
                 self.msg.get('Via'), '%0.1f' % (self.version / 10.0))
+        if extra_response_headers:
+            for header, value in extra_response_headers.items():
+                self.msg[header] = value
```

Comment:
Could replace the for loop with `self.msg.update(extra_response_headers)`

Reply:
``self.msg`` type is ``HTTPMessage`` and not a ``dict``. So, its not possible to do that, sorry.

Reply:
Oh, my mistake